src/mlpack/core/data/load_image_impl.hpp

```diff
@@ -0,0 +1,202 @@
+/**
+ * @file load_image_impl.hpp
+ * @author Mehul Kumar Nirala
+ *
+ * An image loading utillity
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef MLPACK_CORE_DATA_LOAD_IMAGE_IMPL_HPP
+#define MLPACK_CORE_DATA_LOAD_IMAGE_IMPL_HPP
+
+// In case it hasn't been included yet.
+#include "load_image.hpp"
+
+namespace mlpack {
+namespace data {
+
+LoadImage::LoadImage():
+        matrixWidth(-1),
+        matrixHeight(-1),
+        channels(3)
+{
+  fileTypes.clear();
```

Comment:
To avoid code duplication, this could use the second constructor for the initialisation.

Reply:
Okay